athenax-backend/src/main/java/com/uber/athenax/backend/server/jobs/JobWatcherUtil.java

```diff
@@ -150,7 +150,8 @@ static StateView computeState(Map<UUID, JobDefinition> jobs, Map<UUID, InstanceI
 
   static JobDefinitionDesiredstate computeActualState(InstanceInfo info) {
     JobDefinitionResource r = new JobDefinitionResource()
-        .memory(info.status().getAllocatedMB())
+        .memory(info.status().getAllocatedMB() / (info.status().getAllocatedVCores() != 0
```

Comment:
Taking a closer look I don't quite understand this. The memory should be the total amount of memory for the whole job instead of the memory used by each task manager.

Reply:
https://github.com/uber/AthenaX/blob/9de54305f6dc198cbe176f64c8d6dddc6ab2e6ac/athenax-backend/src/main/java/com/uber/athenax/backend/server/yarn/JobConf.java#L48-L55

According to the definition of JobConf, it seems like the amount of memory specified by the Web API parameters is exactly the memory used by each task manager instead of the whole job's memory consumption. And when I start a job with 2 cores and 2G memory, AthenaX starts a job with two TaskManager containers with 1 Core and 2G memory each, so the total amount of memory retrieved from YARN is 4G, which leads to the same issue of continuously restarting.

Maybe this is not the right place to fix this issue ?